### ISSUE TITLE
CTAS with block comment including a function fails parsing

### ISSUE THREAD

**Issue by rubytobi** [sqlfluff/sqlfluff#7914]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Parsing the below SQL fails
```sql
CREATE TABLE table_b
AS (
  SELECT
      col_a
      /*
      min(
      )
      AS type_arr
      */
  FROM table_a
)
```

### Expected Behaviour

I was expecting this to parse fine out of the box.

### Observed Behaviour

When running this, it fails with the below:

```
==== sqlfluff ====
sqlfluff:                4.2.1 python:                3.10.11
implementation:        cpython verbosity:                   3
dialect:                athena templater:               jinja
rules:                                all
== Raw Config:
core:                                   
    dialect:            athena              
    disable_noqa:       False               
    encoding:           autodetect          
    fix_even_unparsable:False               
    ignore:             []                  
    ignore_templated_areas:True                
    large_file_skip_byte_limit:20000               
    large_file_skip_char_limit:0                   
    large_file_skip_fail:False               
    max_line_length:    80                  
    max_parse_depth:    255                 
    max_parse_nodes:    100000              
    output_line_length: 80                  
    processes:          1                   
    render_variant_limit:5                   
    require_dialect:    False               
    rules:              all                 
    runaway_limit:      10                  
    rust_parser_max_iterations:3000000             
    rust_parser_warn_threshold:2000000             
    sql_file_exts:      .sql,.sql.j2,.dml,.ddl,.pkb
    templater:          jinja               
    use_rust_parser:    auto                
    verbose:            3                   
    warn_unused_ignores:False               
    warnings:           []                  
indentation:                            
    ignore_comment_lines:False               
    implicit_indents:   forbid              
    indent_unit:        space               
    indented_ctes:      False               
    indented_joins:     False               
    indented_on_contents:True                
    indented_then:      True                
    indented_then_contents:True                
    indented_using_on:  True                
    skip_implicit_indents_in:case_expression     
    skip_indentation_in:script_content      
    tab_space_size:     4                   
    template_blocks_indent:True                
    trailing_comments:  before              
layout:                                 
    type:                                   
        array_accessor:                         
            spacing_before:     touch:inline        
        array_type:                             
            spacing_within:     touch:inline        
        assignment_operator:                    
            line_position:      leading             
            spacing_within:     touch               
        binary_operator:                        
            line_position:      leading             
            spacing_within:     touch               
        bind_variable:                          
            spacing_within:     touch               
        bracketed_arguments:                    
            spacing_before:     touch:inline        
        casting_operator:                       
            spacing_after:      touch:inline        
            spacing_before:     touch               
        colon:                                  
            spacing_before:     touch               
        colon_delimiter:                        
            spacing_after:      touch               
            spacing_before:     touch               
        column_path_operator:                    
            line_position:      leading             
            spacing_within:     touch               
        comma:                                  
            line_position:      trailing            
            spacing_before:     touch               
        comment:                                
            spacing_after:      any                 
            spacing_before:     any                 
        common_table_expression:                    
            spacing_within:     single:inline       
        comparison_operator:                    
            line_position:      leading             
            spacing_within:     touch               
        dot:                                    
            spacing_after:      touch               
            spacing_before:     touch               
        end_angle_bracket:                      
            spacing_before:     touch               
        end_bracket:                            
            spacing_before:     touch               
        end_of_file:                            
            spacing_before:     touch               
        end_square_bracket:                     
            spacing_before:     touch               
        from_clause:                            
            line_position:      alone               
        function_contents:                      
            spacing_before:     touch:inline        
        function_name:                          
            spacing_within:     touch:inline        
        function_parameter_list:                    
            spacing_before:     touch:inline        
        groupby_clause:                         
            keyword_line_position:leading             
            line_position:      alone               
        having_clause:                          
            keyword_line_position:leading             
            line_position:      alone               
        join_clause:                            
            line_position:      alone               
        limit_clause:                           
            line_position:      alone               
        match_condition:                        
            spacing_within:     touch:inline        
        numeric_literal:                        
            spacing_within:     touch:inline        
        object_reference:                       
            spacing_within:     touch:inline        
        orderby_clause:                         
            keyword_line_position:leading             
            keyword_line_position_exclusions:
window_specification,
aggregate_order_by,
withingroup_clause,
pipe_operator_clause
            line_position:      leading             
        path_segment:                           
            spacing_within:     touch               
        pattern_expression:                     
            spacing_within:     any                 
        placeholder:                            
            spacing_after:      any                 
            spacing_before:     any                 
        select_clause:                          
            line_position:      alone               
        semi_structured_expression:                    
            spacing_before:     touch:inline        
            spacing_within:     touch:inline        
        set_operator:                           
            line_position:      alone:strict        
        sign_indicator:                         
            spacing_after:      touch:inline        
        sized_array_type:                       
            spacing_within:     touch               
        slash:                                  
            spacing_after:      any                 
            spacing_before:     any                 
        slice:                                  
            spacing_after:      touch               
            spacing_before:     touch               
        sql_conf_option:                        
            spacing_within:     touch               
        sqlcmd_operator:                        
            spacing_before:     touch               
        start_angle_bracket:                    
            spacing_after:      touch               
        start_bracket:                          
            spacing_after:      touch               
        start_square_bracket:                    
            spacing_after:      touch               
        statement_terminator:                    
            line_position:      trailing            
            spacing_before:     touch               
        struct_type:                            
            spacing_within:     touch:inline        
        substitution_variable:                    
            spacing_within:     touch               
        template_loop:                          
            spacing_after:      any                 
            spacing_before:     any                 
        tilde:                                  
            spacing_after:      touch:inline        
        typed_array_literal:                    
            spacing_within:     touch               
        typed_struct_literal:                    
            spacing_within:     touch               
        where_clause:                           
            keyword_line_position:leading             
            keyword_line_position_exclusions:pipe_operator_clause
            line_position:      alone               
rules:                                  
    allow_scalar:       True                
    single_table_references:consistent          
    unquoted_identifiers_policy:all                 
    aliasing.column:                        
        aliasing:           explicit            
    aliasing.forbid:                        
        force_enable:       False               
    aliasing.length:                        
    aliasing.table:                         
        aliasing:           explicit            
    aliasing.unused:                        
        alias_case_check:   dialect             
    ambiguous.column_references:                    
        group_by_and_order_by_style:consistent          
    ambiguous.join:                         
        fully_qualify_join_types:inner               
    capitalisation.functions:                    
        extended_capitalisation_policy:consistent          
    capitalisation.identifiers:                    
        extended_capitalisation_policy:consistent          
    capitalisation.keywords:                    
        capitalisation_policy:consistent          
    capitalisation.literals:                    
        capitalisation_policy:consistent          
    capitalisation.types:                    
        extended_capitalisation_policy:consistent          
    convention.blocked_words:                    
        match_source:       False               
    convention.casting_style:                    
        preferred_type_casting_style:consistent          
    convention.count_rows:                    
        prefer_count_0:     False               
        prefer_count_1:     False               
    convention.not_equal:                    
        preferred_not_equal_style:consistent          
    convention.quoted_literals:                    
        force_enable:       False               
        preferred_quoted_literal_style:consistent          
    convention.select_trailing_comma:                    
        select_clause_trailing_comma:forbid              
    convention.terminator:                    
        multiline_newline:  False               
        require_final_semicolon:False               
    elula.qualify_column:                    
        client_name:        democlient          
        disable_resolver:   False               
        resolver_type:      local               
        runid:              "00000000-0000-0000-0000-000000000000"
    elula.wildcard_use:                     
        client_name:        democlient          
        disable_resolver:   False               
        resolver_type:      local               
        runid:              "00000000-0000-0000-0000-000000000000"
    layout.long_lines:                      
        ignore_comment_clauses:False               
        ignore_comment_lines:False               
    layout.newlines:                        
        maximum_empty_lines_between_batches:1                   
        maximum_empty_lines_between_statements:2                   
        maximum_empty_lines_inside_statements:1                   
    layout.select_targets:                    
        single_target_policy:same_line           
        wildcard_policy:    single              
    references.consistent:                    
        force_enable:       False               
    references.from:                        
        force_enable:       False               
    references.keywords:                    
        unquoted_identifiers_policy:aliases             
    references.qualification:                    
        subqueries_ignore_external_references:False               
    references.quoting:                     
        case_sensitive:     True                
        prefer_quoted_identifiers:False               
        prefer_quoted_keyword_style:double_quotes       
        prefer_quoted_keywords:False               
    references.special_chars:                    
        allow_space_in_identifier:False               
        quoted_identifiers_policy:all                 
        unquoted_identifiers_policy:all                 
    structure.join_condition_order:                    
        preferred_first_table_in_join_clause:earlier             
    structure.subquery:                     
        forbid_subquery_in: join                
templater:                              
    unwrap_wrapped_queries:True                
    jinja:                                  
        apply_dbt_builtins: True                

=== [ path: file.sql ] ===

== [file.sql] TEMPLATING
INFO       Rendering String [jinja] (file.sql)
INFO       Slicing File Template
INFO       Rendered 1 variants
== [file.sql] PARSING
INFO       Parse Rendered. Lexing Variant 0
INFO       LEXING RAW (file.sql)
INFO       Lexing file using RsLexer.
INFO       Lexed segments: ['CREATE', ' ', 'TABLE', ' ', 'table_b', '\n', 'AS', ' ', '(', '\n', '  ', 'SELECT', '\n', '      ', 'col_a', '\n', '      ', '/*', '\n', '      ', 'min(', '\n', '      ', ')', '\n', '      ', 'AS type_arr', '\n', '      ', '*/', '\n', '  ', 'FROM', ' ', 'table_a', '\n', ')', '']
INFO       Parse Rendered. Parsing Variant 0
INFO       Using Rust parser (experimental)
INFO       
###
#
# Parsed Tree:
#
###
INFO       
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    unparsable:                                               !! Expected: "<Sequence: [<AnyNumberOf: [<Ref: 'DelimiterGrammar'>..., <Delimited: [<Ref: 'StatementSegment'>]>, <AnyNumberOf:...]>"
        Comments:
[L:  5, P:  7]      |            block_comment:                                    '/*'
[L:  6, P:  7]      |            block_comment:                                    'min('
[L:  7, P:  7]      |            block_comment:                                    ')'
[L:  8, P:  7]      |            block_comment:                                    'AS type_arr'
[L:  9, P:  7]      |            block_comment:                                    '*/'
        Code:
[L:  1, P:  1]      |            word:                                             'CREATE'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            word:                                             'TABLE'
[L:  1, P: 13]      |            whitespace:                                       ' '
[L:  1, P: 14]      |            word:                                             'table_b'
[L:  1, P: 21]      |            newline:                                          '\n'
[L:  2, P:  1]      |            word:                                             'AS'
[L:  2, P:  3]      |            whitespace:                                       ' '
[L:  2, P:  4]      |            start_bracket:                                    '('
[L:  2, P:  5]      |            newline:                                          '\n'
[L:  3, P:  1]      |            whitespace:                                       '  '
[L:  3, P:  3]      |            word:                                             'SELECT'
[L:  3, P:  9]      |            newline:                                          '\n'
[L:  4, P:  1]      |            whitespace:                                       '      '
[L:  4, P:  7]      |            word:                                             'col_a'
[L:  4, P: 12]      |            newline:                                          '\n'
[L:  5, P:  1]      |            whitespace:                                       '      '
[L:  5, P:  9]      |            newline:                                          '\n'
[L:  6, P:  1]      |            whitespace:                                       '      '
[L:  6, P: 11]      |            newline:                                          '\n'
[L:  7, P:  1]      |            whitespace:                                       '      '
[L:  7, P:  8]      |            newline:                                          '\n'
[L:  8, P:  1]      |            whitespace:                                       '      '
[L:  8, P: 18]      |            newline:                                          '\n'
[L:  9, P:  1]      |            whitespace:                                       '      '
[L:  9, P:  9]      |            newline:                                          '\n'
[L: 10, P:  1]      |            whitespace:                                       '  '
[L: 10, P:  3]      |            word:                                             'FROM'
[L: 10, P:  7]      |            whitespace:                                       ' '
[L: 10, P:  8]      |            word:                                             'table_a'
[L: 10, P: 15]      |            newline:                                          '\n'
[L: 11, P:  1]      |            end_bracket:                                      ')'
[L: 11, P:  2]      |    [META] end_of_file:

INFO       Found unparsable segment...
INFO       [L:  1, P:  1]      |unparsable:                                                   !! Expected: "<Sequence: [<AnyNumberOf: [<Ref: 'DelimiterGrammar'>..., <Delimited: [<Ref: 'StatementSegment'>]>, <AnyNumberOf:...]>"
    Comments:
[L:  5, P:  7]      |        block_comment:                                        '/*'
[L:  6, P:  7]      |        block_comment:                                        'min('
[L:  7, P:  7]      |        block_comment:                                        ')'
[L:  8, P:  7]      |        block_comment:                                        'AS type_arr'
[L:  9, P:  7]      |        block_comment:                                        '*/'
    Code:
[L:  1, P:  1]      |        word:                                                 'CREATE'
[L:  1, P:  7]      |        whitespace:                                           ' '
[L:  1, P:  8]      |        word:                                                 'TABLE'
[L:  1, P: 13]      |        whitespace:                                           ' '
[L:  1, P: 14]      |        word:                                                 'table_b'
[L:  1, P: 21]      |        newline:                                              '\n'
[L:  2, P:  1]      |        word:                                                 'AS'
[L:  2, P:  3]      |        whitespace:                                           ' '
[L:  2, P:  4]      |        start_bracket:                                        '('
[L:  2, P:  5]      |        newline:                                              '\n'
[L:  3, P:  1]      |        whitespace:                                           '  '
[L:  3, P:  3]      |        word:                                                 'SELECT'
[L:  3, P:  9]      |        newline:                                              '\n'
[L:  4, P:  1]      |        whitespace:                                           '      '
[L:  4, P:  7]      |        word:                                                 'col_a'
[L:  4, P: 12]      |        newline:                                              '\n'
[L:  5, P:  1]      |        whitespace:                                           '      '
[L:  5, P:  9]      |        newline:                                              '\n'
[L:  6, P:  1]      |        whitespace:                                           '      '
[L:  6, P: 11]      |        newline:                                              '\n'
[L:  7, P:  1]      |        whitespace:                                           '      '
[L:  7, P:  8]      |        newline:                                              '\n'
[L:  8, P:  1]      |        whitespace:                                           '      '
[L:  8, P: 18]      |        newline:                                              '\n'
[L:  9, P:  1]      |        whitespace:                                           '      '
[L:  9, P:  9]      |        newline:                                              '\n'
[L: 10, P:  1]      |        whitespace:                                           '  '
[L: 10, P:  3]      |        word:                                                 'FROM'
[L: 10, P:  7]      |        whitespace:                                           ' '
[L: 10, P:  8]      |        word:                                                 'table_a'
[L: 10, P: 15]      |        newline:                                              '\n'
[L: 11, P:  1]      |        end_bracket:                                          ')'

INFO       Parse Rendered. Variant 0. Lex in 0.0. Parse in 0.01500000013038516.
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    unparsable:                                               !! Expected: "<Sequence: [<AnyNumberOf: [<Ref: 'DelimiterGrammar'>..., <Delimited: [<Ref: 'StatementSegment'>]>, <AnyNumberOf:...]>"
        Comments:
[L:  5, P:  7]      |            block_comment:                                    '/*'
[L:  6, P:  7]      |            block_comment:                                    'min('
[L:  7, P:  7]      |            block_comment:                                    ')'
[L:  8, P:  7]      |            block_comment:                                    'AS type_arr'
[L:  9, P:  7]      |            block_comment:                                    '*/'
        Code:
[L:  1, P:  1]      |            word:                                             'CREATE'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            word:                                             'TABLE'
[L:  1, P: 13]      |            whitespace:                                       ' '
[L:  1, P: 14]      |            word:                                             'table_b'
[L:  1, P: 21]      |            newline:                                          '\n'
[L:  2, P:  1]      |            word:                                             'AS'
[L:  2, P:  3]      |            whitespace:                                       ' '
[L:  2, P:  4]      |            start_bracket:                                    '('
[L:  2, P:  5]      |            newline:                                          '\n'
[L:  3, P:  1]      |            whitespace:                                       '  '
[L:  3, P:  3]      |            word:                                             'SELECT'
[L:  3, P:  9]      |            newline:                                          '\n'
[L:  4, P:  1]      |            whitespace:                                       '      '
[L:  4, P:  7]      |            word:                                             'col_a'
[L:  4, P: 12]      |            newline:                                          '\n'
[L:  5, P:  1]      |            whitespace:                                       '      '
[L:  5, P:  9]      |            newline:                                          '\n'
[L:  6, P:  1]      |            whitespace:                                       '      '
[L:  6, P: 11]      |            newline:                                          '\n'
[L:  7, P:  1]      |            whitespace:                                       '      '
[L:  7, P:  8]      |            newline:                                          '\n'
[L:  8, P:  1]      |            whitespace:                                       '      '
[L:  8, P: 18]      |            newline:                                          '\n'
[L:  9, P:  1]      |            whitespace:                                       '      '
[L:  9, P:  9]      |            newline:                                          '\n'
[L: 10, P:  1]      |            whitespace:                                       '  '
[L: 10, P:  3]      |            word:                                             'FROM'
[L: 10, P:  7]      |            whitespace:                                       ' '
[L: 10, P:  8]      |            word:                                             'table_a'
[L: 10, P: 15]      |            newline:                                          '\n'
[L: 11, P:  1]      |            end_bracket:                                      ')'
[L: 11, P:  2]      |    [META] end_of_file:

==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE TABLE
                       | table_b\nAS (\n  SELECT\n     ...'
WARNING: Parsing errors found and dialect is set to 'athena'. Have you configured your dialect correctly?
==== timings ====
templating:     0.02 lexing:         0.00
parsing:        0.02
==== overall timings ====
Clock time:     0.06
=== templating ===
cnt:               1 sum:            0.02
min:            0.02 max:            0.02
avg:            0.02
=== lexing ===
cnt:               1 sum:            0.00
min:            0.00 max:            0.00
avg:            0.00
=== parsing ===
cnt:               1 sum:            0.02
min:            0.02 max:            0.02
avg:            0.02
```

### How to reproduce

Run `sqlfluff parse --dialect athena file.sql -vvv` and it will fail.
I tried a couple of things:
* Move from block comment to three single line comments, then it works
* Move from `min` to `max` or other functions, still fails
* Move the `min(` part a line up so it's `/* min(` still fails
* Move the `AS type_arr` a line up so it's `) AS type_arr`, then it works
* Move the `)` a line up so it's `min()`, then it works

This indicates to me, that sqlfluff is unnecessarily trying to parse the block comment.

### Dialect

athena

### Version

sqlfluff, version 4.2.1

### Configuration

n/a

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Hi @rubytobi  - thanks for opening this issue.

I have tried, and with the default config I had no issues to parse this statement. I would check your jinja config which might interfere with the SQL. Can you try to `render` the file to see how templating is changing the SQL statement?

**Comment by keraion** [comment]
Ah, I see this is showing up on the Rust side.
